app/services/katello/pulp/smart_proxy_repository.rb

```diff
@@ -0,0 +1,179 @@
+module Katello
+  module Pulp
+    class SmartProxyRepository
+      attr_accessor :smart_proxy
+
+      def initialize(smart_proxy)
+        @smart_proxy = smart_proxy
+      end
+
+      def lifecycle_environments(organization_id = nil)
+        scope = @smart_proxy.lifecycle_environments
+        scope = scope.where(organization_id: organization_id) if organization_id
+        scope
+      end
+
+      def add_lifecycle_environment(environment)
+        @smart_proxy.lifecycle_environments << environment
+      end
```

Comment:
same for this  one, this doesn't need to be in a service, nothing pulp specific.  I kinda even question if this needs to be a method